lib/pos_leader_schedule.go

```diff
@@ -0,0 +1,106 @@
+package lib
+
+import (
+	"crypto/sha256"
+	"github.com/holiman/uint256"
+	"github.com/pkg/errors"
+)
+
+func (bav *UtxoView) GenerateLeaderSchedule() ([]*PKID, error) {
+	// Retrieve CurrentRandomSeedHash.
+	currentRandomSeedHash, err := bav.GetCurrentRandomSeedHash()
+	if err != nil {
+		return nil, errors.Wrapf(err, "UtxoView.GenerateLeaderSchedule: error retrieving CurrentRandomSeedHash: ")
+	}
+
+	// Retrieve top, active validators ordered by stake.
+	validatorEntries, err := bav.GetTopActiveValidatorsByStake(int(bav.Params.LeaderScheduleMaxNumValidators))
```

Comment:
Never use int. Always be clear about the size of the int by using `int32` or `int64`

Reply:
Addressed in: https://github.com/deso-protocol/core/pull/540.